library/psa_crypto.c

```diff
@@ -377,11 +377,7 @@ static void psa_wipe_tag_output_buffer(uint8_t *output_buffer, psa_status_t stat
 /* Key management */
 /****************************************************************/
 
-#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) || \
-    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH)
+#if defined(MBEDTLS_ECP_C)
```

Comment:
Ah, I'm not so sure about that, I was chatting with Ronald and it turns out using `MBEDLTS_xxx_C` macros in PSA files is causing issues with the repo split, so we shouldn't do it. Actually I'm no longer convinced what we did in #7192 is correct. I'll think about it a bit more and come back to you about it.

Reply:
After further chatting with Ronald, it is acceptable for the time being to use `MBEDTLS_xxx_C` feature macros in those parts of PSA that don't have driver support yet, which is the case for ECC derivation, so we're on the right side of the line here after all. Phew!